### PR TITLE
Blanket rename of AuthCodeController to AuthRedirectController

### PR DESCRIPTION
**Pull request by MarcusSmith**
Probably some comment cleanup that's needed, figured it'd be easier to do together here

### CODE REVIEW

File: aqueduct/lib/src/auth/auth_redirect_controller.dart [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+  ///
+  ///         response_type=code&client_id=com.aqueduct.app&state=o9u3jla&username=bob&password=password
+  ///
+  ///
+  /// If not null, [scope] should also be included as an additional form parameter.
+  Future<String> render(AuthRedirectController forController, Uri requestUri,
+      String responseType, String clientID, String state, String scope);
+}
+
+/// [Controller] for issuing OAuth 2.0 authorization codes.
```
Comment (jhdoak) [inline aqueduct/lib/src/auth/auth_redirect_controller.dart]: I think this should include tokens now.

File: aqueduct/lib/src/auth/auth_redirect_controller.dart [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+///
+/// This controller provides an endpoint for the creating an OAuth 2.0 authorization code. This authorization code
```
Comment (jhdoak) [inline aqueduct/lib/src/auth/auth_redirect_controller.dart]: Need to add tokens to this or add another paragraph addressing it.

File: aqueduct/lib/src/auth/auth_redirect_controller.dart [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+  /// Creates a new instance of an [AuthRedirectController].
+  ///
+  /// [authServer] is the required authorization server. If [delegate] is provided, this controller will return a login page for all GET requests.
+  AuthRedirectController(this.authServer, {this.delegate}) {
+    acceptedContentTypes = [
+      ContentType("application", "x-www-form-urlencoded")
+    ];
+  }
+
+  /// A reference to the [AuthServer] used to grant authorization codes.
```
Comment (jhdoak) [inline aqueduct/lib/src/auth/auth_redirect_controller.dart]: and/or tokens.

File: aqueduct/lib/src/auth/auth_redirect_controller.dart [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+
+  /// A randomly generated value the client can use to verify the origin of the redirect.
+  ///
+  /// Clients must include this query parameter and verify that any redirects from this
+  /// server have the same value for 'state' as passed in. This value is usually a randomly generated
+  /// session identifier.
+  @Bind.query("state")
+  String state;
+
+  /// Must be 'code'.
```
Comment (jhdoak) [inline aqueduct/lib/src/auth/auth_redirect_controller.dart]: Must be 'code' or 'token'

File: aqueduct/lib/src/auth/auth_redirect_controller.dart [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+    final renderedPage = await delegate.render(
+        this, request.raw.uri, responseType, clientID, state, scope);
+    if (renderedPage == null) {
+      return Response.notFound();
+    }
+
+    return Response.ok(renderedPage)..contentType = ContentType.html;
+  }
+
+  /// Creates a one-time use authorization code.
```
Comment (jhdoak) [inline aqueduct/lib/src/auth/auth_redirect_controller.dart]: Should mention token as well.

File: aqueduct/lib/src/auth/auth_redirect_controller.dart [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+    if (operation.method == "GET") {
+      return {
+        "200": APIResponse.schema(
+            "Serves a login form.", APISchemaObject.string(),
+            contentTypes: ["text/html"])
+      };
+    } else if (operation.method == "POST") {
+      return {
+        "${HttpStatus.movedTemporarily}": APIResponse(
+            "If successful, the query parameter of the redirect URI named 'code' contains authorization code. "
```
Comment (jhdoak) [inline aqueduct/lib/src/auth/auth_redirect_controller.dart]: Should mention token here.

File: aqueduct/lib/src/auth/auth_redirect_controller.dart [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+    }
+
+    throw StateError("AuthRedirectController documentation failed.");
+  }
+
+  @override
+  Map<String, APIOperation> documentOperations(
+      APIDocumentContext context, String route, APIPath path) {
+    final ops = super.documentOperations(context, route, path);
+    authServer.documentedAuthorizationCodeFlow.authorizationURL =
```
Comment (jhdoak) [inline aqueduct/lib/src/auth/auth_redirect_controller.dart]: This property will need to be renamed.

Comment (joeconwaystk) [reply aqueduct/lib/src/auth/auth_redirect_controller.dart]: I'm not sure in this case - and actually, this creates an interesting question as to how OpenAPI expects this (or if it even accounts for implicit flow?)

**Pull request merged**